main.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    array = []
+    i = 0
+    while i < length
+      array << self[i] if yield(self[i])
+      i += 1
+    end
+    array
+  end
+
+  def my_all?
+    all_true = true
+    i = 0
+    while i < length
+      all_true = false unless yield(self[i])
+      i += 1
+    end
+    all_true
+  end
+
+  def my_any?
```

Comment:
my all should be able to take a parameter 
- [ ]  my_all  should be able should have a guard case to check if a block is given.